test/connectors/util/CustomHttpReaderSpec.scala

```diff
@@ -30,18 +30,22 @@ class CustomHttpReaderSpec extends AnyFreeSpec with Matchers with OptionValues w
 
   val _2xxGenerator: Gen[Int] = Gen.oneOf(Seq(Status.OK, Status.CREATED, Status.ACCEPTED, Status.NO_CONTENT))
   val clientErrorGenerator: Gen[Int] = Gen.oneOf(Seq(Status.LOCKED, Status.FORBIDDEN, Status.NOT_FOUND, Status.BAD_REQUEST))
-  val serverErrorGenerator: Gen[Int] = Gen.oneOf(Seq(Status.INTERNAL_SERVER_ERROR, Status.NOT_IMPLEMENTED, Status.SERVICE_UNAVAILABLE, Status.GATEWAY_TIMEOUT))
+  val serverErrorGenerator: Gen[Int] = Gen.oneOf(Seq(Status.NOT_IMPLEMENTED, Status.SERVICE_UNAVAILABLE, Status.BAD_GATEWAY))
 
   def sut(status: Int) = CustomHttpReader.read("POST", "abc", HttpResponse(status))
 
-  "must convert BAD_GATEWAY to INTERNAL_SERVER_ERROR" in {
-    sut(Status.BAD_GATEWAY).status mustEqual Status.INTERNAL_SERVER_ERROR
+  "must convert BAD_GATEWAY to BAD_GATEWAY" in {
```

Comment:
This should be changed to "must convert GATEWAY_TIMEOUT to BAD_GATEWAY" and the assertion changed.

Reply:
Should be fine now